libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationMetricsWrapper.java

```diff
@@ -216,4 +217,11 @@ private static void updateRouteProgressSessionData(MetricsRouteProgress routePro
   private static Location[] convertToArray(List<Location> locationList) {
     return locationList.toArray(new Location[locationList.size()]);
   }
+
+  private static Date checkNullStartTimestamp(SessionState sessionState) {
+    if (sessionState.startTimestamp() != null) {
```

Comment:
Minor detail: Why not the other way around? Why not using a [guard clause](http://wiki.c2.com/?GuardClause) instead? In the exceptional case (`sessionState.startTimestamp()` is `null`) return `new Date()` and in the normal case return `sessionState.startTimestamp()`. IMO it makes clearer the intention of the method.